### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by arrtemio**

### CODE REVIEW

File: src/features/AuthModal/AuthModal.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        >
+            <Box sx={styles.wrapper} data-testid="AuthModal-wrapper">
+                <Typography
+                    variant="h5"
+                    align="center"
+                    data-testid="AuthModal-title"
+                >
+                    {isRegister
+                        ? t('Registration')
+                        : t('Login')}
```
Comment (sshkatulov) [inline src/features/AuthModal/AuthModal.tsx]: `{t(isRegister ? 'Registration' : 'Login')}`

File: src/shared/lib/helpers/validation/validation.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    return true;
+};
+
+export const passChecking = (pass: string, setter: (message: string) => void) => {
+    if (!pass.trim()) {
+        setter('Field cannot be empty');
+        return false;
+    }
+    if (pass.length < 6) {
+        setter('Password must be more than 6 characters');
```
Comment (sshkatulov) [inline src/shared/lib/helpers/validation/validation.ts]: `< 6` => 'at least 6', not 'more than 6'

File: src/features/AuthModal/AuthModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    );
+
+    test('should renders without error', () => {
+        renderComponent();
+
+        const wrapper = screen.getByTestId('AuthModal-wrapper');
+        const title = screen.getByTestId('AuthModal-title');
+        const switchBtn = screen.getByTestId('AuthModal-caption-switch');
+        const emailField = screen.getByTestId('AuthModal-email');
+        const passField = screen.getByTestId('AuthModal-pass');
```
Comment (sshkatulov) [inline src/features/AuthModal/AuthModal.test.tsx]: you can keep it as is if you preffer but I'm not realy sure what's the point to store them as a separate const

File: src/features/AuthModal/AuthModal.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                    : (
+                        <Button data-testid="AuthModal-btn-login" onClick={handleLogin}>
+                            {t('Login')}
+                        </Button>
+                    )}
+                <Box sx={styles.caption}>
+                    <Typography data-testid="AuthModal-caption" variant="caption">
+                        { isRegister
+                            ? t('Already have an account?')
+                            : t('Do not have an account yet?')}
```
Comment (sshkatulov) [inline src/features/AuthModal/AuthModal.tsx]: `{t(isRegister ? 'Already have an account?' : ''Do not have an account yet?')}`

File: src/features/AuthModal/AuthModal.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                    </Typography>
+                    <Link
+                        variant="caption"
+                        component="button"
+                        onClick={handleSwitchIsRegister}
+                        data-testid="AuthModal-caption-switch"
+                    >
+                        { isRegister
+                            ? t('Login')
+                            : t('Registration')}
```
Comment (sshkatulov) [inline src/features/AuthModal/AuthModal.tsx]: `{t(isRegister ? 'Login' : 'Registration')}`

File: src/pages/TaskList/ui/TaskList.tsx [hunk 6]
```diff
@@ -4,32 +4,35 @@ import { AddTask } from 'features/Addtask/AddTask';
...
 
     const createNewTask = (title: string) => {
         const task = createTaskDto(title);
         dispatch(createTask(task));
     };
 
-    return (
+    return isAuth ? (
         <Container sx={styles.container}>
             <>
```
Comment (sshkatulov) [inline src/pages/TaskList/ui/TaskList.tsx]: should have noticed earlier but this fragment is redunant

File: src/pages/TaskList/ui/TaskList.test.tsx [hunk 7]
```diff
@@ -10,19 +10,26 @@ import { componentRender } from 'shared/lib/tests';
...
     it('renders the task list correctly', async () => {
-        const initialState = {
+        const initialState: StateSchema = {
             tasks: {
                 data: [testTask],
                 isLoading: false,
                 error: undefined,
             },
+            user: {
+                data: testUser,
```
Comment (sshkatulov) [inline src/pages/TaskList/ui/TaskList.test.tsx:29]: could the test for non-logged user be added as well?

Comment (arrtemio) [reply src/pages/TaskList/ui/TaskList.test.tsx:29]: Added a test

File: src/pages/TaskList/ui/TaskList.tsx (hunk 6 above)
Comment (arrtemio) [reply src/pages/TaskList/ui/TaskList.tsx]: Fixed

File: src/features/AuthModal/AuthModal.tsx (hunk 5 above)
Comment (arrtemio) [reply src/features/AuthModal/AuthModal.tsx]: Fixed

File: src/features/AuthModal/AuthModal.tsx (hunk 4 above)
Comment (arrtemio) [reply src/features/AuthModal/AuthModal.tsx]: Fixed

File: src/features/AuthModal/AuthModal.test.tsx (hunk 3 above)
Comment (arrtemio) [reply src/features/AuthModal/AuthModal.test.tsx]: Fixed

File: src/shared/lib/helpers/validation/validation.ts (hunk 2 above)
Comment (arrtemio) [reply src/shared/lib/helpers/validation/validation.ts]: Fixed

File: src/features/AuthModal/AuthModal.tsx (hunk 1 above)
Comment (arrtemio) [reply src/features/AuthModal/AuthModal.tsx]: Fixed

**Review by sshkatulov — APPROVED** [review]

**Pull request merged**